src/main/java/org/jabref/gui/preferences/ShowPreferencesAction.java

```diff
@@ -16,6 +17,12 @@ public ShowPreferencesAction(JabRefFrame jabRefFrame, TaskExecutor taskExecutor)
 
     @Override
     public void execute() {
-        new PreferencesDialogView(jabRefFrame).show();
+        PreferencesDialogView preferencesDialogView = new PreferencesDialogView(jabRefFrame);
+        preferencesDialogView.show();
+        for (PreferencesTab tab: preferencesDialogView.getPreferenceTabList().getItems()) {
```

Comment:
The action shouldn't really care about how the dialog is displayed, that's the job of the dialog itself. Thus, I would prefer if the fix can be moved to the `PreferencesDialogView` class. Maybe to https://github.com/JabRef/jabref/pull/6429/files#diff-39a71e5858d655cd93a5ca0a21ff300aL81 ?

Reply:
Hi, @tobiasdiez 
Thanks for your suggestions. In fact, I tried fix it in the PreferencesDialogView, but the PreferencesTab.getBuilder().getScene()  is not null only when it is initialized after preferencesDialogView.show(). Therefore, this is the only way I found to fix it.  Maybe I missed something important, but I really have no idea to handle that.

Reply:
Where did you tried to add it exactly? The scene should be non-null as soon as the control has been added to the dialog. (But to be honest I don't understand why the preference tab doesn't inherit the css from the preference window).

Reply:
I write some codes to make my point clear:
```
PreferencesDialogView preferencesDialogView = new PreferencesDialogView(jabRefFrame);
for (PreferencesTab tab: preferencesDialogView.getPreferenceTabList().getItems()) {
    System.out.println("result of getBuilder(): ");
    System.out.println(tab.getBuilder());
    System.out.println("result of getScene(): ");
    System.out.println(tab.getBuilder().getScene());
}
preferencesDialogView.show();
System.out.println("after show()");
for (PreferencesTab tab: preferencesDialogView.getPreferenceTabList().getItems()) {
    System.out.println("result of getBuilder(): ");
    System.out.println(tab.getBuilder());
    System.out.println("result of getScene(): ");
    System.out.println(tab.getBuilder().getScene());
}
```
The part of result is:
```
result of getBuilder(): 
	GeneralTabView@5448f579
result of getScene(): 
	null
result of getBuilder(): 
	FileTabView@28189b70
result of getScene(): 
	null

after show()

result of getBuilder(): 
	GeneralTabView@5448f579
result of getScene(): 
	javafx.scene.Scene@454c8139
result of getBuilder(): 
	FileTabView@28189b70
result of getScene(): 
	null

```

You can see that the result of getScene() is not null only after preferencesDialogView.show(). 

By the way, I found that preference tab is not the only one having this problem.
Nearly every pop window suffer this problem like this screenshot:
![捕获](https://user-images.githubusercontent.com/38714962/81243764-4b5fce80-9043-11ea-911b-ac5b9551f4ba.JPG)

Reply:
Hi,
all those popup dialogs inherit from BaseDialog and there is this line for setting the css as well:
`        Globals.getThemeLoader().installCss(getDialogPane().getScene(), Globals.prefs);`

This would explain why those dialogs still have the same size

Reply:
Yes, thank for your reply, I also saw that. However, I can't understand why the codes in the BaseDialog can't work as expected. It really confuse me

Reply:
Thanks for the update. Makes sense that the scene is only initialized after showing the dialog. That's probably also the reason why the line in `BaseDialog` does not work. It could work if you add a listener to the [scene property](https://docs.oracle.com/javase/8/javafx/api/javafx/scene/Node.html#sceneProperty) in order to get notified about the new scene:
`EasyBind.subscribe(getDialogPane().scene(), scene -> installCss(scene, prefs)`.

Reply:
Thankyou for your guidance.
I tried to use
```
EasyBind.subscribe(new ReadOnlyObjectWrapper(getDialogPane().getScene()), scene -> Globals.getThemeLoader().installCss((Scene) scene, Globals.prefs));
```
to replace 
```
Globals.getThemeLoader().installCss(getDialogPane().getScene(), Globals.prefs);
```
in constructor of ```BaseDialog```.
However, I found that this installCss() was still executed before DialogView.show() (this is pseudo code, if I am press "Customize key bindings", then it should be KeyBindingsDialogView().show())
I am confused now about how to solve this.
The only solution I can come up with is adding some codes after each DialogView.show().

Reply:
Hi, you could try to use the properties onShown or onShowing https://docs.oracle.com/javase/8/javafx/api/javafx/scene/control/Dialog.html

Reply:
If you use `getScene` then you access the current value of the scene (which is null) and this is static, i.e. the `ReadOnlyObjectWrapper` will never update. To resolve this, you can use the `sceneProperty` to get notified about the real value.

Something like the following should work:
`EasyBind.wrapNullable(getDialogPane().sceneProperty()).subscribeToValues(scene -> Globals.getThemeLoader().installCss(scene, Globals.prefs))`

Reply:
> If you use `getScene` then you access the current value of the scene (which is null) and this is static, i.e. the `ReadOnlyObjectWrapper` will never update. To resolve this, you can use the `sceneProperty` to get notified about the new value.
> 
> Something like the following should work:
> `EasyBind.wrapNullable(getDialogPane().sceneProperty()).subscribeToValues(scene -> Globals.getThemeLoader().installCss(scene, Globals.prefs))`


I found a interesting thing. That is I found that the KeyBindingsDialogView.getDialogPane().getScene() can be initiated properly before KeyBindingsDialogView.show()

To prove this, I add some codes in construtor of BaseDialog:
```
        System.out.println("before before");
        System.out.println(getDialogPane().getScene());
        Globals.getThemeLoader().installCss(getDialogPane().getScene(), Globals.prefs);
```
and see the behaviours in the debug mode.

following is results:

command line output is :
```
before before
javafx.scene.Scene@b1b581d
07:27:19.064 [JavaFX Application Thread] INFO  org.jabref.gui.util.ThemeLoader - Enabling live reloading of {$workspace}\build\resources\main\org\jabref\gui\Base.css

```
screenshot for debug mode (19 is the font size I set)
![image](https://user-images.githubusercontent.com/38714962/82959972-b02b9a80-9feb-11ea-8edc-0fd06b03af35.png)

I think it means Themeloader.installCss() can apply to KeyBindingsDialogView properly but the font size does not change. I am comfused again about this.

Reply:
Is it also the same scene that is used after the dialog is shown? Maybe it's registering to the old scene that gets replaced?